src/app/components/query-explainer/query-explainer.component.ts

```diff
@@ -0,0 +1,129 @@
+    public blockHashSystemTimes: string[] = [];
+
+    ngOnInit(): void {
+        this.sqlQuery = this.extractSqlQueryFromRoute();
+
+        if (this.sqlQuery) {
+            const blockHashObservables$: Observable<string>[] = [];
+            this.queryExplainerService
+                .proccessQuery(this.sqlQuery)
+                .pipe(
+                    tap((response) => {
+                        this.sqlQueryExplainer = response;
+                        response.input.datasets
+                            .map((dataset) => ({ datasetId: dataset.id, blockHash: dataset.blockHash }))
+                            .forEach(({ datasetId, blockHash }) => {
+                                blockHashObservables$.push(
+                                    this.blockService.requestSystemTimeBlockByHash(datasetId, blockHash),
+                                );
+                            });
+                    }),
+                    switchMap((response: QueryExplainerResponse) => {
+                        return this.queryExplainerService.verifyQuery(response);
+                    }),
+                    catchError((e: HttpErrorResponse) => {
+                        return of(e.error as VerifyQueryResponse);
+                    }),
+                    tap((data) => {
+                        this.verifyResponse = data;
+                    }),
+                    switchMap(() => forkJoin(blockHashObservables$)),
+                    tap((blockHashSystemTimes) => {
+                        this.blockHashSystemTimes = blockHashSystemTimes;
+                    }),
+                    takeUntilDestroyed(this.destroyRef),
+                )
+                .subscribe(() => this.cdr.detectChanges());
+        }
+    }
+
+    public copyToClipboard(event: MouseEvent, text: string): void {
+        this.clipboard.copy(text);
+        if (event.currentTarget !== null) {
+            const currentElement: HTMLButtonElement = event.currentTarget as HTMLButtonElement;
+            const currentElementChildren: HTMLCollectionOf<HTMLElement> =
+                currentElement.children as HTMLCollectionOf<HTMLElement>;
+            setTimeout(() => {
+                currentElementChildren[0].style.display = "inline-block";
+                currentElementChildren[1].style.display = "none";
+                currentElement.classList.remove("clipboard-btn--success");
+            }, AppValues.LONG_DELAY_MS);
+            currentElementChildren[0].style.display = "none";
+            currentElementChildren[1].style.display = "inline-block";
+            currentElement.classList.add("clipboard-btn--success");
+        }
+    }
+
+    private extractSqlQueryFromRoute(): MaybeNull<string> {
+        const sqlQueryFromRoute: MaybeUndefined<string> = this.activatedRoute.snapshot.queryParams[
+            ProjectLinks.URL_QUERY_PARAM_SQL_QUERY
+        ] as MaybeUndefined<string>;
+        return sqlQueryFromRoute ?? "";
+    }
```

Comment:
How it can be null, if you have a default value?

Reply:
Fixed